lib/rubygems/commands/push_command.rb

```diff
@@ -100,5 +123,22 @@ def send_gem name
     with_response response
   end
 
+  private
+
+  def get_hosts_for(name)
+    gem_data = Gem::Package.new(name)
+
+    default_gem_server = nil
+    if gem_data.spec.metadata.has_key?("default_gem_server")
+      default_gem_server = gem_data.spec.metadata["default_gem_server"]
+    end
+
+    push_host = nil
+    if gem_data.spec.metadata.has_key?("allowed_push_host")
+      push_host = gem_data.spec.metadata["allowed_push_host"]
+    end
+
+    [default_gem_server, push_host]
+  end
```

Comment:
I hope I didn't misunderstood the code in this method, but wouldn't the
following be equivalent?
``` ruby
gem_metadata = Gem::Package.new(name).spec.metadata

[
  gem_metadata["default_gem_server"],
  gem_metadata["allowed_push_host"]
]
```

Reply:
yes, this is much better, thanks!

Reply:
lol, specs brake with this